packages/platform-ios/src/commands/runIOS/index.ts

```diff
@@ -359,7 +363,15 @@ function bootSimulator(selectedSimulator: Device) {
   }
 }
 
+function getTargetBuildDir(buildSettings: string) {
+  const targetBuildMatch = /TARGET_BUILD_DIR = (.+)$/m.exec(buildSettings);
+  return targetBuildMatch && targetBuildMatch[1]
+    ? targetBuildMatch[1].trim()
+    : null;
```

Comment:
this broke the build for me because we have multiple targets in the project, and this line matched the incorrect target.

Reply:
cc @janicduplessis

Reply:
Looks like xcodebuild has a -target flag, maybe we can try setting that.

Reply:
Yep. Looking quickly, we'd need to introduce it as a new flag to `run-ios`

Reply:
Yea not sure how that worked before with the hardcoded path. @leethree How did you handle building with multiple targets before this change? Which target gets selected now and which got selected before this change?

Reply:
I think Xcode just put the resulted app in `${appName}.app` and hardcoded path worked fine before.

We have targets for the iOS app itself, and some watchkit and widget stuff in our build scheme, which means there are multiple build settings in whatever `xcodebuild -showBuildSetting` outputs.

If I understand correctly, this script will match the first `TARGET_BUILD_DIR` in the output, which is not necessarily the app target we want.

Reply:
Okay, I'm able to get the output for a specific target using `xcodebuild -project projectName.xcodeproj -sdk iphonesimulator -configuration Debug -showBuildSettings -target targetName`

I wonder if it would be safe to assume the target name is the same name as scheme, seems like a scheme with the same name as the target is created when you create a new target. We could use that as the default and allow customizing the target with a new option maybe in case the name is different.

Reply:
@leethree In your case does your scheme and target have the same name?

Reply:
no we don't have the same name for scheme and target. I guess a more reliable way is to match all `TARGET_BUILD_DIR` and find the one that actually contains `${appName}.app`.